tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -75,19 +81,30 @@ void ExploreBranch(TTree &t, std::set<std::string> &bNamesReg, ColumnNames_t &bN
       if (!prefix.empty())
          newPrefix = fullName + ".";
 
-      ExploreBranch(t, bNamesReg, bNames, subBranch, newPrefix, friendName);
+      ExploreBranch(t, bNamesReg, bNames, subBranch, newPrefix, friendName, foundLeaves, allowDuplicates);
 
-      if (t.GetBranch(fullName.c_str()))
-         UpdateList(bNamesReg, bNames, fullName, friendName);
-      else if (t.GetBranch(subBranchName.c_str()))
-         UpdateList(bNamesReg, bNames, subBranchName, friendName);
+      if (t.GetBranch(fullName.c_str())) {
+         TLeaf *leaf = t.GetBranch(fullName.c_str())->GetLeaf(fullName.c_str());
+         const bool canAdd = allowDuplicates ? true : !(leaf != nullptr && ContainsLeaf(foundLeaves, leaf));
+         if (canAdd) {
+            UpdateList(bNamesReg, bNames, fullName, friendName);
+            foundLeaves.insert(leaf);
+         }
+      } else if (t.GetBranch(subBranchName.c_str())) {
+         TLeaf *leaf = t.GetBranch(subBranchName.c_str())->GetLeaf(fullName.c_str());
+         const bool canAdd = allowDuplicates ? true : !(leaf != nullptr && ContainsLeaf(foundLeaves, leaf));
+         if (canAdd) {
+            UpdateList(bNamesReg, bNames, subBranchName, friendName);
+            foundLeaves.insert(leaf);
+         }
+      }
```

Comment:
I chose not to use an helper function as its signature would have been:
`Function(std::set<std::string> &bNamesReg, ColumnNames_t &bNames, std::string &branchName, std::string &friendName, std::set<TLeaf *> &foundLeaves, const TLeaf* leaf))`
And it's a bunch of parameters to just save 3 lines, which I personally don't like. 
But if you think it's better, I'll do it

Reply:
the goal is not to save 3 lines, but to avoid repetition of logic. because when people will change that logic somewhere they will forget to change it everywhere else, and that's not fun to debug.

If it's possible then please add the helper function. If you don't modify the strings you take as arguments you can take them as `std::string_view` instead of `const std::string&`, and you don't need the `const` in `const TLeaf *leaf`.

Reply:
Ok! 
What do you think if instead of creating a new helper I push this logic into UpdateList? 
It makes sense to me, it updates the list if necessary, otherwise does nothing.